### PR TITLE
Home page : with the last SDK functions

### PR DESCRIPTION
**Pull request by Le-Caignec**
- For Abbes : 

- [x] Ticket 43 : https://iexecproject.atlassian.net/browse/PRO-43?atlOrigin=eyJpIjoiZTI3MjQ2ZjUwMTk1NGMxOTg2ZmY1YWQ4ZjRmNmIxNDYiLCJwIjoiaiJ9. You should review the `./src/protectedData/ProtectedData.tsx`

- For Pierre : 

- [x] Ticket 41 : https://iexecproject.atlassian.net/browse/PRO-41?atlOrigin=eyJpIjoiYTRhMTkwNTc4ODE4NDRmMThkOWMzYWIzOTYzYWEwZjAiLCJwIjoiaiJ9. You should review `./src/naviguate/Navigate.tsx`
- [x] Ticket 44: https://iexecproject.atlassian.net/browse/PRO-44?atlOrigin=eyJpIjoiYzQ3N2UzMDM1M2Y0NDI2ZWI3MGFkMTI3ODI4M2VmNWMiLCJwIjoiaiJ9. You should review `./src/newProtectedData/NewProtectedData.tsx` and app folder (in particular `appSlice.ts`file)

NB: The web3modal includes all the available wallet today. As we didn't decided clearly which wallet we want to expose for our products, I didn't restricted yet the web3modal for a certain range of wallets.

- For Seddik : 

- [x] Ticket 42: https://iexecproject.atlassian.net/browse/PRO-42?atlOrigin=eyJpIjoiZmRkMzI5ZmFjNzEzNDgwYjg5Yjg0MTI4NDgwZmI1NTciLCJwIjoiaiJ9. You should review `./src/components/ProtectedDataCard.tsx`, may be`./src/protectedData/ProtectedData.tsx` and app folder (in particular `appSlice.ts`file)

If you have more specific remark on other files, I'm open to it. I think improves are possible around the RTK. But it could be done in other tickets I think

### CODE REVIEW

**Review by abbesBenayache — APPROVED** [review]

**Review by abbesBenayache — APPROVED** [review]

File: src/features/protectedData/ProtectedData.tsx [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
-      dataType: 'Profile',
-      id: '0x126561e41f561c872f033a80d6eaecd964b6f0d6',
-    },
-    {
-      title: 'Date of Birth',
-      date: '18/01/2022',
-      dataType: 'Document',
-      id: '0x8ece0c27c52237329aa61f521f77632454754e3c',
-    },
+  const ITEMS_PER_PAGE = 8;
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/features/protectedData/ProtectedData.tsx]: Commit: 44c02a6b956965135971cbdbfdef55a219d44424

[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/features/protectedData/ProtectedData.tsx]: Yeah it's a good idea, I will talk about that to Alexandre and improve mock-up if he  agree

File: src/app/api.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { IExecDataProtector } from '@iexec/dataprotector';
+import { connect, getAccount } from 'wagmi/actions';
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/app/api.ts]: Commit: b4054235751cb1d2860438a48df2191fde45646e

File: package.json [hunk 3]
```diff
@@ -1,54 +1,44 @@
...
     "@mui/x-data-grid": "^6.0.4",
-    "@testing-library/jest-dom": "^5.16.5",
-    "@testing-library/react": "^13.4.0",
-    "@testing-library/user-event": "^13.5.0",
-    "@types/jest": "^27.5.2",
-    "@types/node": "^16.18.22",
-    "@types/react": "^18.0.31",
-    "@types/react-dom": "^18.0.11",
+    "@reduxjs/toolkit": "^1.9.5",
     "@wagmi/connectors": "^0.3.10",
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply package.json:21]: no it didn't work. It raises issue without this package

File: src/features/protectedData/ProtectedData.tsx [hunk 4]
```diff
@@ -54,43 +54,45 @@ export default function ProtectedData() {
               <NewProtectedDataButton />
             </Box>
           </Box>
-          <Box sx={{ mx: 4 }}>
+          <Box sx={{ mx: 4, paddingBottom: 20 }}>
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/features/protectedData/ProtectedData.tsx:56]: - done : Commit: 5893a240686db897ec29f2605f6125055c0c9af7
Good idea it enables to remove an import and make the code easier to understand

File: package.json [hunk 5]
```diff
@@ -1,54 +1,44 @@
 {
-  "name": "data-protector-use-case",
-  "version": "0.1.0",
+  "name": "vite-project",
   "private": true,
+  "version": "0.0.0",
+  "type": "module",
+  "scripts": {
+    "dev": "vite",
```
Comment (PierreJeanjacquot) [inline package.json:7]: README is outdated, remove everything related to cra and add basic commands for vite

File: package.json [hunk 6]
```diff
@@ -1,54 +1,44 @@
 {
-  "name": "data-protector-use-case",
-  "version": "0.1.0",
+  "name": "vite-project",
```
Comment (PierreJeanjacquot) [inline package.json:2]: change name

File: src/app/api.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+export const api = createApi({
+  baseQuery: fetchBaseQuery(),
+  endpoints: () => ({}),
+});
+
+export const getIExecDataProtectorAndRefresh =
+  async (): Promise<IExecDataProtector> => {
+    try {
+      const result = await connect({
+        connector: new InjectedConnector() as any,
```
Comment (PierreJeanjacquot) [inline src/app/api.ts:15]: wagmi is configured for a bunch of providers but you only use `InjectedConnector` it seems buggy

File: public/vite.svg
Comment (PierreJeanjacquot) [inline public/vite.svg:1]: should remove

File: src/app/appSlice.ts [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+      queryFn: async (owner) => {
+        try {
+          const data = await iExecDataProtector?.fetchProtectedData({ owner });
+          return { data: data };
+        } catch (e: any) {
+          return { error: e.message };
+        }
+      },
+    }),
+    createPotectedData: builder.mutation<
```
Comment (PierreJeanjacquot) [inline src/app/appSlice.ts:79]: typo `createProtectedData`

File: src/app/appSlice.ts [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+      queryFn: async (args) => {
+        try {
+          const data = await iExecDataProtector?.protectData(args);
+          return { data: data };
+        } catch (e: any) {
+          return { error: e.message };
+        }
+      },
+    }),
+    fetchGrantedAcces: builder.query<GrantedAccess[], string>({
```
Comment (PierreJeanjacquot) [inline src/app/appSlice.ts:92]: typo `fetchGrantedAccess`

File: src/features/navigate/Navigate.tsx [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+  Button,
+  Tab,
+  Tabs,
+  Toolbar,
+  Typography,
+} from '@mui/material';
+import { useEffect, useState } from 'react';
+import { Outlet, useNavigate } from 'react-router-dom';
+import { useAccount, useDisconnect } from 'wagmi';
+import { useWeb3Modal } from '@web3modal/react';
```
Comment (PierreJeanjacquot) [inline src/features/navigate/Navigate.tsx:15]: how `wagmi` is supposed to work with `web3modal`?

File: src/features/navigate/Navigate.tsx [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+    }
+    if (value === 'sendMail') {
+      navigate('/sendMail');
+    }
+  }, [value]);
+
+  return (
+    <div>
+      <AppBar position="static" elevation={0} id="appbar">
+        <Toolbar id="tootBar">
```
Comment (PierreJeanjacquot) [inline src/features/navigate/Navigate.tsx:60]: tootBar ?

File: src/features/navigate/Navigate.tsx [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+    setValue(newValue);
+  };
+
+  useEffect(() => {
+    if (isDisconnected) {
+      open();
+    } else if (isConnected && !isAccountConnected) {
+      dispatch(initDataProtector());
+    }
+  }, [isDisconnected, isAccountConnected]);
```
Comment (PierreJeanjacquot) [inline src/features/navigate/Navigate.tsx:46]: React Hook useEffect has missing dependencies: 'dispatch', 'isConnected', and 'open'. Either include them or remove the dependency array. eslint [react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)

File: src/features/navigate/Navigate.tsx [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+
+  useEffect(() => {
+    if (value === 'myProtectedData') {
+      navigate('/protectedData');
+    }
+    if (value === 'sendMail') {
+      navigate('/sendMail');
+    }
+  }, [value]);
```
Comment (PierreJeanjacquot) [inline src/features/navigate/Navigate.tsx:55]: React Hook useEffect has a missing dependency: 'navigate'. Either include it or remove the dependency array. eslint [react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)

File: src/features/navigate/Navigate.tsx [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { shortAddress } from '../../utils/utils';
+
+export default function Navigate() {
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const { open } = useWeb3Modal();
+  const { address, isConnected, isDisconnected } = useAccount();
+  const { disconnect } = useDisconnect();
+  const [value, setValue] = useState('myProtectedData');
```
Comment (PierreJeanjacquot) [inline src/features/navigate/Navigate.tsx:30]: rename this variable with an explicit name

File: src/features/navigate/Navigate.tsx [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+  }, [value]);
+
+  return (
+    <div>
+      <AppBar position="static" elevation={0} id="appbar">
+        <Toolbar id="tootBar">
+          <img
+            onClick={() => navigate('/protectedData')}
+            src={img}
+            alt="The immage can't be loaded"
```
Comment (PierreJeanjacquot) [inline src/features/navigate/Navigate.tsx:64]:
```suggestion
            alt="The image can't be loaded"
```

File: package.json (hunk 5 above)
Comment (Le-Caignec) [reply package.json:7]: Commit: ddc01e703a3fc87e0fcb5b96e03f608eef7104df

File: public/vite.svg
Comment (Le-Caignec) [reply public/vite.svg:1]: Commit: 16c5d9b3e14c28dba82de5cc5d8807bdcc37dff7

File: src/app/appSlice.ts (hunk 8 above)
Comment (Le-Caignec) [reply src/app/appSlice.ts]: Commit: e3a9faaf6d5dd6bed1d462805d2432c351ec1af0

File: src/features/navigate/Navigate.tsx (hunk 11 above)
Comment (Le-Caignec) [reply src/features/navigate/Navigate.tsx]: Commit: 84c6866dc8d7174825acaab45661db3b0469ccd2

File: src/features/navigate/Navigate.tsx (hunk 12 above)
Comment (Le-Caignec) [reply src/features/navigate/Navigate.tsx]: Commit: d212097b52245eb8a883d455be861ffc8cd8352f

File: src/features/navigate/Navigate.tsx [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+  const { open } = useWeb3Modal();
+  const { address, isConnected, isDisconnected } = useAccount();
+  const { disconnect } = useDisconnect();
+  const [value, setValue] = useState('myProtectedData');
+
+  //get the state from the store
+  const isAccountConnected = useAppSelector(selectAppIsConnected);
+  const loading = useAppSelector(selectThereIsSomeRequestPending);
+
+  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/features/navigate/Navigate.tsx]: Commit: 82d73eaf21c9ffd46ada6a312cb8d600e7f3ca71

File: src/features/navigate/Navigate.tsx (hunk 14 above)
Comment (Le-Caignec) [reply src/features/navigate/Navigate.tsx]: Commit: edfe3a605d4b4cf3335981f9d6e19ca1b9062f69

File: src/components/ProtectedDataCard.tsx [hunk 17]
```diff
@@ -1,26 +1,35 @@
...
 }
 
 export default function ProtectedDataCard(props: ProtectedDataProps) {
-  const naviguate = useNavigate()
+  const navigate = useNavigate();
   return (
     <Card
       sx={{ minWidth: 260, cursor: 'pointer' }}
-      onClick={() => naviguate(`/protectedData/consent/${props.id}`)}
+      onClick={() => navigate(`/protectedData/consent/${props.id}`)}
```
Comment (SeddikBellamine) [inline src/components/ProtectedDataCard.tsx:18]: change navigation to constants: example : navigate(PAGE1)

File: src/components/ProtectedDataCard.tsx [hunk 18]
```diff
@@ -1,26 +1,35 @@
...
     >
       <CardContent id="cardContent">
         <Box
           sx={{ height: '60px', display: 'flex', justifyContent: 'flex-end' }}
         >
-          <Chip id="chipLabel" label={props.dataType} size="small" />
+          <Chip
+            id="chipLabel"
+            label={
+              (hasKey(props.schema, 'email') && 'Email') ||
```
Comment (SeddikBellamine) [inline src/components/ProtectedDataCard.tsx:27]: i didn't understand why you used both haskey() and 'Email ?

Comment (Le-Caignec) [reply src/components/ProtectedDataCard.tsx:27]: I make that to know if the keys email or file is present in the schema of the protected Data to display it if it's the case

File: src/features/navigate/Navigate.tsx (hunk 10 above)
Comment (Le-Caignec) [reply src/features/navigate/Navigate.tsx:15]: the link between the two library is made in the main.tsx
``

File: src/app/api.ts (hunk 7 above)
Comment (Le-Caignec) [reply src/app/api.ts:15]: Commit: 7bf8b83d30c9ea974e744bb7483f3b85797f3e8c

**Review by PierreJeanjacquot — COMMENTED** [review]
WIP review

File: src/features/newProtectedData/NewProtectedData.tsx [hunk 19]
```diff
@@ -12,101 +12,84 @@ import {
...
+        };
+        fileReader.readAsArrayBuffer(file);
+      });
     } else {
-      return Promise.reject(new Error('No file selected'))
+      return Promise.reject(new Error('No file selected'));
     }
   }
 
   const handleSubmit = async (event: any) => {
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/features/newProtectedData/NewProtectedData.tsx]: commit 82d73eaf21c9ffd46ada6a312cb8d600e7f3ca71

File: src/features/newProtectedData/NewProtectedData.tsx [hunk 20]
```diff
@@ -12,101 +12,84 @@ import {
...
+    setFilePath(event.target.value);
+    setFile(event.target.files?.[0]);
+  };
   const handleNameChange = (event: any) => {
-    setName(event.target.value)
-  }
+    setName(event.target.value);
+  };
 
   async function create_ArrayBuffer(file?: File): Promise<ArrayBuffer> {
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/features/newProtectedData/NewProtectedData.tsx]: Commit: ac2f00e715da20e5bb68137356dea034294e28e4

File: src/features/newProtectedData/NewProtectedData.tsx [hunk 21]
```diff
@@ -12,101 +12,84 @@ import {
...
-        //const ProtectedDataAddress = await createCNFT(data!, name)
-        setCNftAddress('0x123456789')
-        setError('')
-      } catch (error) {
-        setError(String(error))
-        setCNftAddress('')
-      }
-      setLoading(false)
+    if (dataType && name && ((isValidEmail && email) || file)) {
+      await createPotectedData({ data, name });
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/features/newProtectedData/NewProtectedData.tsx]: commit e3a9faaf6d5dd6bed1d462805d2432c351ec1af0

File: src/app/api.ts (hunk 7 above)
Comment (abbesBenayache) [reply src/app/api.ts]: in api.ts : 
- 'IExecDataProtector' is defined but never used
- 'getAccount' is declared but its value is never read.ts(6133)

Comment (Le-Caignec) [reply src/app/api.ts]: done : Commit: 35d4b6b533bf3c97a5bab6e32f31df32c61bfc1f

**Review by abbesBenayache — APPROVED** [review]

**Pull request merged**